src/Microsoft.AspNet.Diagnostics/StatusCodePagesExtensions.cs

```diff
@@ -0,0 +1,133 @@
+        /// <returns></returns>
+        public static StatusCodePagesOptions WithHandler(this StatusCodePagesOptions options, Func<StatusCodeContext, Task> handler)
+        {
+            options.HandleAsync = handler;
+            return options;
+        }
+
+        /// <summary>
+        /// Specifies the response body to send. This may include a '{0}' placeholder for the status code.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="contentType"></param>
+        /// <param name="bodyFormat"></param>
+        /// <returns></returns>
+        public static StatusCodePagesOptions WithResponse(this StatusCodePagesOptions options, string contentType, string bodyFormat)
+        {
+            return options.WithHandler(context =>
+            {
+                var body = string.Format(CultureInfo.InvariantCulture, bodyFormat, context.HttpContext.Response.StatusCode);
+                return context.HttpContext.Response.SendAsync(body, contentType);
+            });
+        }
+
+        /// <summary>
+        /// Specifies that responses should be handled by redirecting with the given location url template.
+        /// This may include a '{0}' placeholder for the status code. Ulrs starting with '~' will have PathBase prepeneded,
+        /// where any other url will be used as is.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="locationFormat"></param>
+        /// <returns></returns>
+        public static StatusCodePagesOptions WithRedirect(this StatusCodePagesOptions options, string locationFormat)
+        {
+            if (locationFormat.StartsWith("~"))
+            {
+                locationFormat = locationFormat.Substring(1);
+                return options.WithHandler(context =>
+                {
+                    var location = string.Format(CultureInfo.InvariantCulture, locationFormat, context.HttpContext.Response.StatusCode);
+                    context.HttpContext.Response.Redirect(context.HttpContext.Request.PathBase + location);
+                    return Task.FromResult(0);
+                });
+            }
+            else
+            {
+                return options.WithHandler(context =>
+                {
+                    var location = string.Format(CultureInfo.InvariantCulture, locationFormat, context.HttpContext.Response.StatusCode);
+                    context.HttpContext.Response.Redirect(location);
+                    return Task.FromResult(0);
+                });
+            }
+        }
+
+        /// <summary>
+        /// Specifies an alternate middleware pipeline to execute to generate the response body.
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="configuration"></param>
+        /// <returns></returns>
+        public static StatusCodePagesOptions WithTangent(this StatusCodePagesOptions options, Action<IApplicationBuilder> configuration)
```

Comment:
I'm not sure if I was in the design meetings for this, but this name is... weak :frowning: I'd rather even call it something like `WithPipeline` or something because that's at least closer to what it is.